compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -2870,6 +2870,20 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             );
                         }
                     }
+
+                    if let ty::RawPtr(..) = base_t.kind() && idx_t.is_integral() {
+                        let base_snip = self.tcx.sess.source_map().span_to_snippet(base.span);
+                        let idx_snip = self.tcx.sess.source_map().span_to_snippet(idx.span);
+                        if let Ok(base_snip) = base_snip && let Ok(idx_snip) = idx_snip {
+                            err.span_suggestion(
+                                expr.span,
+                                "consider using `wrapping_add` or `add` for indexing into raw pointer",
+                                format!("{base_snip}.wrapping_add({idx_snip})"),
```

Comment:
This should be a multipart suggestion, this would avoid `source_map` calls
- `(base.span.between(idx.span), ".wrappig_add(")`
- `(idx.span.shrink_to_hi(), ")")`

(and similarly below)